internal/azure/environments.go

```diff
@@ -0,0 +1,152 @@
+package internal
```

Comment:
If we are going to keep this we should make sure we import the License/copyright. I think this is just a copy over from autorest?

Reply:
nit: I would suggest changing this package name. I think the convention is to name the package the same as the directory. If we don't want to use `package azure` as the name then maybe something like `package azureenv`?

Reply:
That is a good call! I was also unsure about the best approach to move forward with this. I couldn't find an equivalent to the types described in `go-autorest/autorest/azure/environments.go`, and since the types and functions have no dependencies on external libraries or API calls, I thought maybe having an internal copy of it might get us past the dependency on `go-autorest/autorest/azure`. I am also fine falling back on the autorest if that seems like an option, but that would cause autorest libraries to be pulled in as a dependency in this repo.

At the very least I shall import the license and copyright until have a confirmed way we want to move forward with this 🙏🏼

Reply:
Update: I think we can use this from the new sdk instead of hardcoding these ourselves: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/cloud/cloud.go

Reply:
The [vault auth API](https://developer.hashicorp.com/vault/api-docs/auth/azure#environment) mentions the following values for `environment`: AzurePublicCloud, AzureUSGovernmentCloud, AzureChinaCloud, AzureGermanCloud. It looks like the names have changed in the new sdk so maybe we will want to make sure we have backwards compat as well?

Reply:
@fairclothjm thanks for finding this! I think we are able to use this package, and I have made the updates in the latest commit. Here are some caveats from this:

- First, since the updated SDK no longer has an `Environment` type, I opted to update our Azure config API to deprecate `environment` and use `cloud_config` instead since now we are technically using `cloud.Configuration` instances. Alternatively, we can keep the API field name as is, `environment`, even though we are using `cloud.Configuration` in order to have a much smaller impact radius (though in that case there might be confusion in the way the field is named)

- Second, we would need to explicitly list the valid values for the new `cloud_config` option, which are `AZURECHINACLOUD`, `AZUREPUBLICCLOUD` and `AZUREGOVERNMENTCLOUD`

Let me know what you think of these changes!

Reply:
If it were me I would keep it as `environment`. I don't see much benefit to changing it.

Reply:
+1 to keeping this as `environment`. The cost of deprecation and having clients migrate to the new field outweighs the benefit of naming consistency IMO.